www/js/survey/enketo-survey-service.js

```diff
@@ -105,25 +105,26 @@ angular.module('emission.enketo-survey.service', [
   /**
    * _restoreAnswer restore the most recent answer for the survey
    * @param {EnketoAnswer[]} answers survey answers
-   * @returns {string} answer string
+   * @returns {Promise<string>} answer string promise
    */
   function _restoreAnswer(answers) {
-    let answer = null;
-    answers = EnketoSurveyAnswer.filterByNameAndVersion(_state.name, answers);
-    if (!answers.length) {
-      return null;
-    }
-    if (_state.opts.trip) {
-      answer = DiaryHelper.getUserInputForTrip(_state.opts.trip, undefined, answers);
-      if (answer) {
-        return answer.data.xmlResponse;
+    return EnketoSurveyAnswer.filterByNameAndVersion(_state.name, answers).then(answers => {
+      let answer = null;
+      if (!answers.length) {
+        return null;
       }
-    }
-    if (_state.opts.uuid) {
-      answers = answers.filter(answer => answer.data.user_uuid === _state.opts.uuid);
-      answer = answers.length ? answers[0] : null;
-    }
-    return answer ? answer.data.xmlResponse : null;
+      if (_state.opts.trip) {
+        answer = DiaryHelper.getUserInputForTrip(_state.opts.trip, undefined, answers);
+        if (answer) {
+          return answer.data.xmlResponse;
+        }
```

Comment:
When the survey is launched, user can launch by adding `opts.trip`, `opts.uuid` or without any options at all.
If they use `opts.trip`, then it tries to restore answer based on the trip `start_ts` and `end_ts`.
If they use `opts.uuid`, then it tries to restore answer based on `data.uuid`.
If none `opts` is given, we fall back to line `126`.

Reply:
I understand that.

However, the code here is

```
        if (answer) {
          return answer.data.xmlResponse;
        }
```

The code at line 126 is 

```
       return answer ? answer.data.xmlResponse : null;
```

They both seem to have the same functionality, so I was just wondering why you used two different code fragments. It is not a huge issue, (which is why it is labeled "nit"), but I like to follow standard patterns if possible.

Reply:
If the answer is found by checking with `trip`, I just wanted to return right away and not checking for `uuid` any further. I could not return `null` if answer is not found when checking against `trip` since we can further check on `uuid`. That is why two bits are slightly different.

Reply:
I think we are speaking at cross-purposes here, but this is a nit, so I am going to let it go.